src/Nethermind/Nethermind.Evm.Test/EvmObjectFormatTests.cs

```diff
@@ -0,0 +1,581 @@
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using FluentAssertions;
+using Nethermind.Core.Extensions;
+using Nethermind.Core.Specs;
+using Nethermind.Specs;
+using Nethermind.Core.Test.Builders;
+using NUnit.Framework;
+using Nethermind.Specs.Forks;
+using NSubstitute;
+using Nethermind.Evm.CodeAnalysis;
+using Nethermind.Blockchain;
+using static Nethermind.Evm.CodeAnalysis.ByteCodeValidator;
+using NUnit.Framework.Constraints;
+using System;
+using System.Collections;
+using DotNetty.Common.Utilities;
+using Org.BouncyCastle.Asn1.Utilities;
+using static Nethermind.Evm.Test.Eip3541Tests;
+using System.Collections.Generic;
+using System.Linq;
+using Nethermind.Evm.Tracing;
+using Nethermind.Evm.Tracing.GethStyle;
+
+namespace Nethermind.Evm.Test
+{
+    /// <summary>
+    /// https://gist.github.com/holiman/174548cad102096858583c6fbbb0649a
+    /// </summary>
+    public class EvmObjectFormatTests : VirtualMachineTestsBase
+    {
+        protected override long BlockNumber => MainnetSpecProvider.ShanghaiBlockNumber;
+        byte[] Classicalcode(byte[] bytecode, byte[] data = null)
+        {
+            var bytes = new byte[(data is not null && data.Length > 0 ? data.Length : 0) + bytecode.Length];
+
+            Array.Copy(bytecode, 0, bytes, 0, bytecode.Length);
+            if (data is not null && data.Length > 0)
+            {
+                Array.Copy(data, 0, bytes, bytecode.Length, data.Length);
+            }
+
+            return bytes;
+        }
+        byte[] EofBytecode(byte[] bytecode, byte[] data = null)
+        {
+            var bytes = new byte[(data is not null && data.Length > 0 ? 10 + data.Length : 7) + bytecode.Length];
+
+            int i = 0;
+
+            // set magic
+            bytes[i++] = 0xEF; bytes[i++] = 0x00; bytes[i++] = 0x01;
+
+            // set code section
+            var lenBytes = bytecode.Length.ToByteArray();
+            bytes[i++] = 0x01; bytes[i++] = lenBytes[^2]; bytes[i++] = lenBytes[^1];
+
+            // set PushData section
+            if (data is not null && data.Length > 0)
+            {
+                lenBytes = data.Length.ToByteArray();
+                bytes[i++] = 0x02; bytes[i++] = lenBytes[^2]; bytes[i++] = lenBytes[^1];
+            }
+            bytes[i++] = 0x00;
+
+            // set the terminator byte
+            Array.Copy(bytecode, 0, bytes, i, bytecode.Length);
+            if (data is not null && data.Length > 0)
+            {
+                Array.Copy(data, 0, bytes, i + bytecode.Length, data.Length);
+            }
+
+            return bytes;
+        }
+
+        // valid code
+        [TestCase("0xEF00010100010000", true, 1, 0, true)]
```

Comment:
In EIP test cases we have more examples https://eips.ethereum.org/EIPS/eip-3540#test-cases. Could we move all of them to our tests?

Reply:
I added them